custom_components/tuya_local/devices/qoto_05_water_valve.yaml

```diff
@@ -0,0 +1,66 @@
+name: QOTO 05 Water Valve
+primary_entity:
+  entity: number
+  icon: "mdi:pipe-valve"
+  dps:
+    - id: 2
+      type: integer
+      name: value
+      unit: "%"
+      range:
+        min: 0
+        max: 100
+      mapping:
+        - step: 1
+secondary_entities:
+  - entity: sensor
+    category: diagnostic
+    icon: "mdi:valve"
+    name: Current Valve Position
+    dps:
+      - id: 3
+        name: sensor
+        type: integer
+        unit: "%"
+  - entity: sensor
+    category: diagnostic
+    name: Auto Shutoff Time Remaining
+    icon: "mdi:timer"
+    dps:
+      - id: 101
+        name: sensor
+        type: integer
+        unit: s
+  - entity: number
+    category: config
+    name: Auto Shutoff Timer
+    icon: "mdi:timer"
+    dps:
+      - id: 11
+        type: integer
+        name: value
+        unit: s
```

Comment:
Copied all of these from the other QOTO device (QOTO 03 Sprinkler) which looks like it was quite old and not following these conventions.

Having a little issue with this one though. I too think that having it in seconds is a bit bizarre. Will break it down into the two timers.

**Shutoff Timer**
The device itself allows setting it with second accuracy (not sure when/ why that would be important but I figured if someone wants to water their plant for 1 minute and 1 second why should I stop them. I hate that it displays in HA as pure seconds rather than HH:MM:SS format but not sure if there is a way to do this.

**Time Remaining**
The device has a special way of doing this. For time > 60s it only updates every minute (presumably to save battery power). For time < 60s it updates every 10 seconds to provide more granularity.
Is there a way with scaling to preserve this granularity when less than 60s (was thinking of using float but that seems potentially fraught with disaster).